ratelimits/gcra_test.go

```diff
@@ -0,0 +1,180 @@
+package ratelimits
+
+import (
+	"testing"
+	"time"
+
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/config"
+	"github.com/letsencrypt/boulder/test"
+)
+
```

Comment:
Here's a test for `divThenRound` just in-case upstream `math.Round` ever changes for any reason ever.
```
func Test_divThenRound(t *testing.T) {
	var zero = int64(0)
	var one = int64(1)
	var two = int64(2)
	var three = int64(3)
	var int64MinVal = int64(-9223372036854775808)

	// This is a special case from math.Round
	r := divThenRound(zero, zero)
	test.AssertEquals(t, r, int64MinVal)

	r = divThenRound(one, zero)
	test.AssertEquals(t, r, int64MinVal)

	r = divThenRound(zero, one)
	test.AssertEquals(t, r, zero)

	// Just dividing, no rounding
	r = divThenRound(one, one)
	test.AssertEquals(t, r, one)

	// Round up
	r = divThenRound(one, two)
	test.AssertEquals(t, r, one)

	// Round down
	r = divThenRound(one, three)
	test.AssertEquals(t, r, zero)
}
```

Reply:
We don't anticipate this changing and if it did it would indicate that there is or was a bug in the standard library.